python/lsst/ts/MTMount/mtmount_csc.py

```diff
@@ -434,33 +444,56 @@ async def connect(self):
             self.monitor_telemetry_client()
         )
 
-    def connect_callback(self, communicator):
-        self.evt_connected.set_put(
-            command=communicator.client_connected, replies=communicator.server_connected
+    async def _wait_for_mock_controller(self):
+        """Wait for the mock controller to be running.
+
+        Returns
+        -------
+        command_port : `int`
+            The command port.
+        telemetry_port : `int`
+            The telemetry port.
+        """
+        # Sample data:
+        # Mock TMA controller running: command_port=40827, telemetry_port=33133
+        running_regex = re.compile(
+            r"running: command_port=(\d+), telemetry_port=(\d+)$"
         )
-        if self.should_be_connected and not communicator.connected:
-            self.should_be_connected = False
-            self.fault(
-                code=enums.CscErrorCode.CONNECTION_LOST,
-                report="Lost connection to low-level controller",
-            )
+        while True:
+            if self.mock_controller_process.returncode is not None:
+                raise RuntimeError("Mock controller process failed")
+            line_bytes = await self.mock_controller_process.stdout.readline()
+            line_str = line_bytes.decode().strip()
+            self.log.debug("Mock controller: %s", line_str)
+            match = running_regex.search(line_str)
+            if match is not None:
+                return int(match[1]), int(match[2])
 
     async def disconnect(self):
         """Disconnect from the low-level controller.
 
-        Close and delete the communicator, if present,
-        and stop the mock controller process, if running.
+        Close the connection to the mock controller, if connected.
+        Sop the mock controller and telemetry process, if running.
```

Comment:
`Sop` should probably be `Stop`.

Reply:
Good catch. And the text is wrong (outdated information); this now only stops the telemetry process. The CSC now starts the mock controller when the CSC is constructed and stops it at exit.